File: app/src/main/java/com/bitflaker/lucidsourcekit/charts/FrequencyData.java
package com.bitflaker.lucidsourcekit.charts;

public class FrequencyData {
    private final float frequency;
    private final float frequencyTo;
    private final float duration;

    public FrequencyData(float frequency, float duration) {
        this.frequency = frequency;
        this.frequencyTo = Float.NaN;
        this.duration = duration;
    }

    public FrequencyData(float frequencyFrom, float frequencyTo, float duration) {
        this.frequency = frequencyFrom;
        this.frequencyTo = frequencyTo;
        this.duration = duration;
    }

    public float getDuration() {
        return duration;
    }

    public float getFrequency() {
        return frequency;
    }

    public float getFrequencyTo() {
        return Float.isNaN(frequencyTo) ? frequency : frequencyTo;
    }

    public float getFrequencyStepSize(int sampleRate) {
        return (getFrequencyTo() - frequency) / (duration * sampleRate);
    }
}

File: app/src/main/java/com/bitflaker/lucidsourcekit/data/FrequencyData.kt
package com.bitflaker.lucidsourcekit.data

import kotlin.Int

class FrequencyData {
    val frequency: Float
    val duration: Float
    private val frequencyTo: Float

    constructor(frequency: Float, duration: Float) {
        this.frequency = frequency
        this.frequencyTo = Float.NaN
        this.duration = duration
    }

    constructor(frequencyFrom: Float, frequencyTo: Float, duration: Float) {
        this.frequency = frequencyFrom
        this.frequencyTo = frequencyTo
        this.duration = duration
    }

    fun getFrequencyTo(): Float {
        return if (frequencyTo.isNaN()) frequency else frequencyTo
    }

    fun getFrequencyStepSize(sampleRate: Int): Float {
        return (getFrequencyTo() - frequency) / (duration * sampleRate)
    }
}
